### ISSUE TITLE
Scoping ActionController::Parameters doesn't work when passing a custom policy with `with:`

### ISSUE THREAD

**Issue by louim** [palkan/action_policy#70]
Hey!

I came across a potential problem when scoping params with `authorized_scope`, when trying to use a specific policy, for example: 
```
class CourseContentsController
  def create
    ...
    authorized_scope(params.require(:course_content), with: CoursePolicy)
    ...
  end
end
```
Will raise a `ActionController::UnfilteredParameters: unable to convert unpermitted parameters to hash` error.

Also related, but maybe not a bug, but the default `implicit_authorization_target` injected here: 
https://github.com/palkan/action_policy/blob/72804730c0ceca718a8effe9d38c9efdd283facb/lib/action_policy/rails/controller.rb#L54-L56
will output `Couldn't find policy class for nil` when unable to safe_constantize the controller. Maybe that is not possible, but I think it would be helpful for the message to read `Couldn't find policy class for CourseContent` or maybe a different message since it is not able to find a matching class?

I created a PR with failing tests to better demonstrate the issue.

**Comment by palkan** [comment]
Fixed and released in 0.3.1.

**Comment by louim** [comment]
Thank you! 🎉
